### PR TITLE
Update CohortMembership create view and serializer to accept list uploads

### PR DESCRIPTION
**Pull request by aphilipharrell**
### Fixes Part 1/2 of https://github.com/academic-innovation/frontend-app-mogc-partners/issues/53

- Update `CohortMembershipCreateView` to add cohort and user to context
- Add `CohortMembershipListSerializer` to handle bulk inserts
- Update `CohortMembershipSerializer` to use list serializer if `data` is a list
- Remove `catalog` endpoints now that they are fully replace in frontend app

For use in https://github.com/academic-innovation/frontend-app-mogc-partners/pull/74.

### CODE REVIEW

**Comment by aphilipharrell** [comment]
[Partner portal improvements discovered during AAUM pilot](https://trello.com/c/Im60O5yf/750-partner-portal-improvements-discovered-during-aaum-pilot)

File: tests/test_views.py [hunk 1]
```diff
@@ -480,6 +482,25 @@ def test_only_own_cohort(self, api_rf):
...
+        request = api_rf.post(
+            f"/memberships/{cohort.uuid}/",
+            json.dumps(user_data),
+            content_type="application/json",
+        )
+        force_authenticate(request, manager.user)
+
+        response = member_create_view(request, cohort_uuid=cohort.uuid)
+
+        assert response.status_code == 201
```
Comment (michaelwheeler) [inline tests/test_views.py:502]: Should we also assert that the expected number of memberships were created?

File: mogc_partnerships/views.py [hunk 2]
```diff
@@ -172,24 +171,28 @@ class CohortMembershipCreateView(generics.CreateAPIView):
...
-        User = get_user_model()
-        try:
-            membership_account = User.objects.get(email=email)
-        except User.DoesNotExist:
-            membership_account = None
-        serializer.save(cohort=cohort, user=membership_account)
+
+        return {"cohort": cohort, "user": user}
+
+    def perform_create(self, serializer):
```
Comment (michaelwheeler) [inline mogc_partnerships/views.py:194]: Does `perform_create` even need an implementation here? I think this is the same as what is inherited.

File: mogc_partnerships/views.py [hunk 3]
```diff
@@ -172,24 +171,28 @@ class CohortMembershipCreateView(generics.CreateAPIView):
...
             raise PermissionDenied("No")
-        email = serializer.validated_data.get("email")
-        User = get_user_model()
-        try:
-            membership_account = User.objects.get(email=email)
-        except User.DoesNotExist:
-            membership_account = None
-        serializer.save(cohort=cohort, user=membership_account)
+
+        return {"cohort": cohort, "user": user}
```
Comment (michaelwheeler) [inline mogc_partnerships/views.py:192]: I'm not seeing how `user` is being referenced in the serializers. Does it need to be included in the context?

**Review by michaelwheeler — CHANGES_REQUESTED** [review]
Hey @aphilipharrell sorry it's taken so long to get this reviewed. I added a few questions and suggestions, but hopefully nothing too outrageous.

File: mogc_partnerships/serializers.py [hunk 4]
```diff
@@ -89,6 +91,36 @@ def get_is_enrolled(self, obj):
...
+        ]
+
+        objects = models.CohortMembership.objects.bulk_create(
+            cohort_memberships, ignore_conflicts=True
+        )
+        # bulk_create doesn't return autoincremented IDs with MySQL DBs
+        # so we have to query results separately
+        return models.CohortMembership.objects.filter(
+            email__in=[cm.email for cm in objects]
+        ).all()
```
Comment (michaelwheeler) [inline mogc_partnerships/serializers.py:121]: Should this also filter the QuerySet to only include memberships associated with `cohort`?

Also, I think the `.all()` is not necessary.

**Review by michaelwheeler — CHANGES_REQUESTED** [review]
Sorry @aphilipharrell just one more question/comment to address.

**Review by michaelwheeler — APPROVED** [review]
Thanks @aphilipharrell 🎉

**Pull request merged**